### ISSUE TITLE
Adding tests for TypeScript

### ISSUE THREAD

**Issue by flying-sheep** [eslint/eslint#7817]
Hi, I’d like to extend a few rules so that they play well with the [typescript-eslint-parser](https://github.com/eslint/typescript-eslint-parser).

Is it feasible to pull that parser in for tests, or should I go another route (e.g. serializing the SourceCode emitted by it and using that)?

---

An example would be the [new-parens](http://eslint.org/docs/rules/new-parens) rule and the follwing code:

```ts
new Storage<RootState>('state')
```

[Currently that rule works](https://github.com/eslint/eslint/blob/master/lib/rules/new-parens.js#L70-L75) by skipping all `)` tokens after the `callee` of a `NewExpression` and then checks if the next token is a `(` and still part of the `NewExpression`. I’d like to extend this so it works with TypeScript Syntax (idk how to do that best yet) and would like to have tests validating that it works.

**Comment by nzakas** [comment]
If you can do it in such a way that the rule isn't looking for TypeScript-specific syntax, then that's fine (we just don't want special cases for nonstandard syntax embedded in the rules).

**Comment by flying-sheep** [comment]
i think i can do that. my main question however was how to write the tests:

can I use (and therefore add a dev dependency for) typescript-eslint-parser?

**Comment by platinumazure** [comment]
I think we ran into a similar thing with babel-eslint. Looking at our package.json devDependencies, I don't see babel-eslint in there. So I would guess that fixtures with parse data were created for those test cases. Not 100% sure, but my guess is you should use fixtures as well.

**Comment by not-an-aardvark** [comment]
For an example, see https://github.com/eslint/eslint/commit/c0f49378f2d43c83065f85d5adff25ac24a9dc82

**Comment by flying-sheep** [comment]
so if both is OK, I’ll just see what looks most natural and go for it.

**Comment by nzakas** [comment]
Sounds good. I'm going to close this issue as the question was answered. We will look forward to reviewing your PRs.
